src/logging.cpp

```diff
@@ -0,0 +1,104 @@
+#include "logging.hpp"
+
+#include "config.h"
+#include "util/files.hpp"
+
+#include <quill/Quill.h>
+
+#include <filesystem>
+#include <string>
+
+#if defined(__CYGWIN__) || defined(__MINGW32__) || defined(__MINGW64__)                \
+    || defined(QUILL_NO_THREAD_NAME_SUPPORT)
+static const std::string LOGLINE_FORMAT =
+    "%(ascii_time) [%(thread:<6)] [%(fileline:<18)] %(level_name) [%(logger_name:<14)] "
+    "- %(message)";
+#else
+static const std::string LOGLINE_FORMAT =
+    "%(ascii_time) [%(thread:<16):%(thread_name:<16)] [%(fileline:<18)] %(level_name) "
+    "[%(logger_name:<14)] - %(message)";
+
+#endif
+
+namespace logging {
+
+using namespace quill;
+using cc = quill::ConsoleColours;
+
+LogLevel _LOG_LEVEL;
+
+void
+init(quill::LogLevel log_level, bool structured)
```

Comment:
`structured` is unused. Is there are reason for this?

Reply:
I wasn't done yet